packages/react-dom/src/__tests__/ReactCompositeComponent-test.js

```diff
@@ -122,8 +129,13 @@ describe('ReactCompositeComponent', () => {
       }
 
       const el = document.createElement('div');
-      expect(() => {
-        expect(() => ReactDOM.render(<Child test="test" />, el)).toThrow(
+      const root = ReactDOMClient.createRoot(el);
+      await expect(async () => {
+        await expect(async () => {
+          await act(() => {
+            root.render(<Child test="test" />);
+          });
+        }).toThrow(
```

Comment:
I think you'll need to do `rejects.toThrow` to wait for the promise. toErrorDev does this differently than the built in jest matcher. I don't know it actually works to do both toThrow and toErrorDev with an async expect though

Reply:
I just had to handle toThrow here: https://github.com/facebook/react/pull/27995/files#diff-bad4915321a8bed0f2496dfcc5340439847fe9aafff205edc0b83ab579a889c4R68

Reply:
@kassens another thing you can do for test like this that isn't specifically testing batching, is to use `flushSync` and then the expect can stay sync:

```js
expect(() => {
  ReactDOM.flushSync(() => {
    root.render(<Child test="test" />);
  });
).toThrow(...)
```